datafusion/datasource-parquet/src/source.rs

```diff
@@ -794,6 +822,34 @@ impl FileSource for ParquetSource {
     fn schema_adapter_factory(&self) -> Option<Arc<dyn SchemaAdapterFactory>> {
         self.schema_adapter_factory.clone()
     }
+
+    /// When push down to parquet source of a sort operation is possible,
+    /// create a new ParquetSource with reverse_scan enabled.
+    ///
+    /// # Phase 1 Behavior (Current)
+    /// Returns `Inexact` because we're only reversing the scan direction and reordering
+    /// files/row groups. We still need to verify ordering at a higher level.
+    ///
+    /// # Phase 2 (Future)
+    /// Could return `Exact` when we can guarantee that the scan order matches the requested order, and
+    /// we can remove any higher-level sort operations.
+    ///
+    /// TODO support more policies in addition to reversing the scan.
+    fn try_pushdown_sort(
+        &self,
+        _order: &[PhysicalSortExpr],
+    ) -> datafusion_common::Result<SortOrderPushdownResult<Arc<dyn FileSource>>> {
+        // Note: We ignore the specific `order` parameter here because the decision
+        // about whether we can reverse is made at the FileScanConfig level.
+        // This method creates a reversed version of the current ParquetSource,
+        // and the FileScanConfig will reverse both the file list and the declared ordering.
+        let new_source = self.clone().with_reverse_scan_inexact(true);
```

Comment:
I don't think we should rely on what `FileScanConfig` does.
I think we should do our own analysis here and return what we are able to do.
Then `FileScanConfig` should combine that with what it is able to do.